dart_application_2/bin/exceptions.dart

```diff
@@ -0,0 +1,27 @@
+class EmptyStringException implements Exception {
+  static String errMsg() => "Please enter non empty value";
+}
+
+class NonNumericException implements Exception {
+  static String errMsg() => "Value entered should be numeric";
+}
+
+class NonNaturalException implements Exception {
+  static String errMsg() => "Value entered should be natural number";
+}
+
+class NonWholeException implements Exception {
+  static String errMsg() => "Value entered should be natural number";
+}
+
+class OutOfRangeException implements Exception {
+  static String errMsg() => "Value entered is out of range";
+}
+
+class AlphabetException implements Exception {
+  static String errMsg() => "Name should only contain alphabets";
+}
+
+class CaseException implements Exception {
+  static String errMsg() => "First letter of name should be capital";
+}
```

Comment:
Not a bad way but can be better.
By using Inheritance here, its easier to create each exception.

For example:

```
abstract class CollegeAppException implements Exception{

  final String errorMessage;

  CollegeAppException(this.errorMessage);

  @override
  String toString() {
    return errorMessage;
  }
}

class IllegalTypeException extends CollegeAppException{
  IllegalTypeException() : super('Illegal Type Exception');
}

class NumberException extends CollegeAppException{
  NumberException() : super('LegalException Exception');
}
```

And usage: 

```
try{
    throw IllegalTypeException();
  } on CollegeAppException catch(exception,stackTrace){
    print(exception.toString());
    print(stackTrace);
  }
```

This way you can avoid catching each of them separately and catch the base class pointer